internal/exec/exec.go

```diff
@@ -208,8 +209,47 @@ func execFieldSelection(ctx context.Context, r *Request, s *resolvable.Schema, f
 			if f.field.ArgsPacker != nil {
 				in = append(in, f.field.PackedArgs)
 			}
+
+			// Before hook directive visitor
+			if len(f.field.Directives) > 0 {
+				for _, directive := range f.field.Directives {
+					if visitor, ok := r.Visitors[directive.Name.Name]; ok {
+						var values = make([]interface{}, 0)
+						for _, inValue := range in {
+							values = append(values, inValue.Interface())
+						}
+
+						if visitorErr := visitor.Before(ctx, directive, values); err != nil {
+							err := errors.Errorf("%s", visitorErr)
+							err.Path = path.toSlice()
+							err.ResolverError = visitorErr
+							return err
+						}
+					}
+				}
+			}
```

Comment:
These visitors will only be invoked if we use method resolvers. How about when we use struct fields as resolvers?

Reply:
I'm talking about [lines 263 to 269 below](https://github.com/graph-gophers/graphql-go/pull/446/files#diff-87ea4cfcdd6bb8062908727b36ef05d305ddfbd5ed89322e2276f315ac63dd1bR263-R269)